app/src/main/java/com/sdpteam/connectout/event/viewer/EventMapViewFragment.java

```diff
@@ -0,0 +1,72 @@
+package com.sdpteam.connectout.event.viewer;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageButton;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.SupportMapFragment;
+import com.google.android.gms.maps.model.MarkerOptions;
+import com.sdpteam.connectout.R;
+import com.sdpteam.connectout.event.Event;
+
+public class EventMapViewFragment extends Fragment implements OnMapReadyCallback {
+
+    private static final int DEFAULT_MAP_ZOOM = 15;
+    private final EventViewModel eventViewModel;
+    private GoogleMap map;
+
+    public EventMapViewFragment(EventViewModel eventViewModel) {
+        this.eventViewModel = eventViewModel;
+        eventViewModel.refreshEvent();
+    }
+
+    @Nullable
+    @Override
```

Comment:
do you think refactoring would be necessary here ?

Reply:
I agree, as both map fragments functions in the same way.
I refactored `LocationPicker` too since they are all Maps fragments